rasa/shared/utils/validation.py

```diff
@@ -247,6 +247,18 @@ def validate_training_data_format_version(
             raise TypeError
 
         if version.parse(LATEST_TRAINING_DATA_FORMAT_VERSION) >= parsed_version:
+            if version_value < LATEST_TRAINING_DATA_FORMAT_VERSION:
+                # not raising here since it's not critical
+                logger.warning(
```

Comment:
I would suggest raising a `UserWarning` using  `rasa.shared.utils.io.raise_warning` (according to the convention [here](https://www.notion.so/rasa/Python-Code-Conventions-26835ffd72484dd699f28fa693916d2e#6d4bea523bef4b18b200f47988087d4e)).

Reply:
I considered that, but then the output wasn't coloured. So I used the convention used on this line:
https://github.com/RasaHQ/rasa/blob/main/rasa/shared/utils/validation.py#L233
but am happy to update.

Reply:
I guess it depends on the shell you're using, I'm using `zsh` and UserWarnings appear bright yellow or orange 😄

Reply:
Sorry, meant to say this: https://ohmyz.sh/